### PR TITLE
docs: Edge Cloud-init reference page

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR adds details about the cloud-init stages that are available to users when applying customization to the edge host installation. 
💻 [Preview](https://deploy-preview-1153--docs-spectrocloud.netlify.app/clusters/edge/)

- [Cloud Init Stages](https://deploy-preview-1153--docs-spectrocloud.netlify.app/clusters/edge/install/cloud-init)
- [Stylus Configuration page](https://deploy-preview-1153--docs-spectrocloud.netlify.app/clusters/edge/install/stylus-reference)

🎫 [DOC-731](https://spectrocloud.atlassian.net/jira/software/c/projects/DOC/boards/62?modal=detail&selectedIssue=DOC-731)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1153#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (cloudmaniac) [inline content/docs/04-clusters/03-edge/30-cloud-init.md:62]: Maybe have those 2 as bullet points to really delineate the usage?

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (cloudmaniac) [inline content/docs/04-clusters/03-edge/30-cloud-init.md:58]: I would complete this section by adding that "all user-data is preserved to final stage." as Rishi mentioned

**Review by cloudmaniac — COMMENTED** [review]
I left 2 minor comments. One other thing that I was wondering is how the system behaves if different config for the same user-data field are competing?

Otherwise it looks fantastic! I learned something today

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/03-edge/30-cloud-init.md:58]: @cloudmaniac I do not disagree with your point, but help me understand here. How does that help the user, or in what context would the user need that information? I'm mainly trying to identify if the information would be noise or if there is an actual need.

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/03-edge/30-cloud-init.md:62]: Great feedback, and I agree. It looks cleaner in bullet form.

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/03-edge/30-cloud-init.md]: @cloudmaniac How would they compete? If you use, for example, the stage `boot` the installer and the OS pack, each stage will run when applicable. Whatever you do in the installer stage will happen in the installer stage, and what you do at the OS stage happens when the OS is booting up.

Comment (cloudmaniac) [reply content/docs/04-clusters/03-edge/30-cloud-init.md]: For example if an admin create 2 incompatible/competing configuration in the same stage (e.g., `fs` or `network`), what happens?

Comment (antongisli) [reply content/docs/04-clusters/03-edge/30-cloud-init.md]: it is worth noting (and please check with Rishi), that any configuration you execute in a later stage could overrwrite something done in an earlier stage. But of course if there are different boots (installer, OS) both running cloud init, then they would be like two separate namespaces and could only overwrite what had happened in their own namespaces.

**Comment by spectro-prow** [comment]
@karl-cardenas-coding: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
librarium_build | 6c4ca5279bad719f0aab9b3cc8640f7421993b7c | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_librarium/1153/librarium_build/1632948457448673280) | `/test librarium_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: content/docs/04-clusters/03-edge/30-cloud-init.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/03-edge/30-cloud-init.md]: @rishi-anand any chance we could get your input on the discussion above?

**Review by 3pings — APPROVED** [review]
This looks great

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1339819855" title="Approved">3pings</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1339819855" title="Approved">3pings</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-3/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/03-edge/20-advanced.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/20-advanced.md:6]:
```suggestion
fullWidth: false
hiddenFromNav: true
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md:210]:
```suggestion
3. Update the sample site provided with your site names and locations. Make the required changes in the **pages/index.js** file. The **readme** file provides additional details about the files to be changed and instructions on how to build and test the application locally.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md:225]:
```suggestion
8. Provide the URL of the deployed app to the Edge Installer user data. Use the `stylus.site.registrationURL` parameter.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md:239]:
```suggestion
10. Power on the Edge host device and scan the QR code.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md:266]:
```suggestion
The next step in the installation process is to add the Edge host to a cluster or to create an Edge Native host cluster. Check out the [Create Cluster Definition](/clusters/edge/site-deployment/site-installation/cluster-deployment) guide to complete the last step of the installation process.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:16]:
```suggestion
To complete the Edge Installation process, an Edge host must become a member of a host cluster. You can add an Edge host to an existing host cluster of type Edge Native, or you can create a new host cluster for Edge hosts and add the Edge host to it.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:42]:
```suggestion
5. Give the cluster a name, description, and tags. Click on **Next**.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:48]:
```suggestion
7. Review your cluster profile values and make changes as needed. Click on **Next**.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:51]:
```suggestion
8.  Provide the host cluster with the Virtual IP (VIP) address used by the physical site. You can also select any SSH keys in case you need to remote into the host cluster. And you can provide a list of Network Time Protocol (NTP) servers. Click on **Next**.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:54]:
```suggestion
9. The node configuration page is where you can specify what Edge hosts make up the host cluster. Assign Edge hosts to the **master-pool** and the **worker-pool**. When you have completed configuring the node pools, click on **Next**.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md
Comment (ritawatson) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/07-cluster-deployment.md:57]:
```suggestion
10. The Settings page is where you can configure a patching schedule, security scans, backup settings, and set up Role-Based Access Control (RBAC). Review the settings and make changes if needed. Click on **Validate**.
```

**Review by ritawatson — APPROVED** [review]
Some suggestions. Otherwise a very good PR.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1339819855" title="Approved">3pings</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1344490282" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-3/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ritawatson — APPROVED** [review]
Some suggestions. Otherwise a very good PR.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1339819855" title="Approved">3pings</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1153#pullrequestreview-1345265341" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-3/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/03-edge.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge.md:41]:
```suggestion
Edge is still in active development and is subject to change. Review the Palette [release notes](/release-notes) for updates and changes.

```

File: content/docs/04-clusters/03-edge/02-edge-configuration/30-cloud-init.md
Comment (karl-cardenas-coding) [reply content/docs/04-clusters/03-edge/02-edge-configuration/30-cloud-init.md:85]: Nice catch

File: content/docs/04-clusters/03-edge/09-site-deployment.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment.md:53]:
```suggestion
| K8s Version |- 1.24.x<br/>- 1.25.x<br/>- 1.26.x |
```

File: content/docs/04-clusters/03-edge/09-site-deployment/03-installer.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/03-installer.md:19]:
```suggestion

<br />
```

File: content/docs/04-clusters/03-edge/09-site-deployment/04-stage.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/04-stage.md:205]:
```suggestion
10. Choose a cluster that has access to the datastore used for storing the VMDK. Choose the Datastore where VMDK is stored.
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md:61]:
```suggestion
- Access to network information about the physical site, specifically the network Virtual IP Address (VIP).
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md:128]:
```suggestion
Perform the following steps to proceed with the installation at the site in your VMware environment.

<br / >
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md:153]:
```suggestion
6. The last step is to create a cluster if you don't have a host cluster that the Edge host can join. Follow the steps in the [Create Cluster Definition](/clusters/edge/site-deployment/site-installation/cluster-deployment) to complete the site installation. 
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation.md:157]:
```suggestion
Once the cluster is created, the installation process will continue. The Palette Edge Host agent will download all required artifacts and reboot the Edge host. 
```

File: content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/09-site-deployment/05-site-installation/06-edge-host-registration.md:37]:
```suggestion
You can automate the registration process by using registration tokens. This method requires providing the registration token in the user data. If a default project was selected for the registration token, that is the project the Edge host will be registered under. You can override the default project by providing a project id in the user data.
```

**Pull request merged**
